From 307202b74568b0bd0bd8d445264a8430d26cbc6f Mon Sep 17 00:00:00 2001
From: Milan <me@petabyteboy.de>
Date: Tue, 24 Apr 2018 11:38:08 +0200
Subject: [PATCH] nspawn: fix copy/reflink fallback when directory already
 exists (#8781)

(cherry picked from commit 679def2a0644d4ffeb8561ac78bcc175cfd2aa25)
---
 src/basic/btrfs-util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/basic/btrfs-util.c b/src/basic/btrfs-util.c
index 3d30497f74..0db8292e70 100644
--- a/src/basic/btrfs-util.c
+++ b/src/basic/btrfs-util.c
@@ -1703,7 +1703,7 @@ int btrfs_subvol_snapshot_fd(int old_fd, const char *new_path, BtrfsSnapshotFlag
                 if (r == -ENOTTY && (flags & BTRFS_SNAPSHOT_FALLBACK_DIRECTORY)) {
                         /* If the destination doesn't support subvolumes, then use a plain directory, if that's requested. */
                         if (mkdir(new_path, 0755) < 0)
-                                return r;
+                                return -errno;
 
                         plain_directory = true;
                 } else if (r < 0)
-- 
2.17.0

